src/lib/pubkey/ec_group/ec_point.h

```diff
@@ -378,17 +378,20 @@ std::pair<BigInt, BigInt> BOTAN_UNSTABLE_API OS2ECP(const uint8_t data[], size_t
                                                     const BigInt& curve_b);
 
 template<typename Alloc>
-PointGFp OS2ECP(const std::vector<uint8_t, Alloc>& data, const CurveGFp& curve)
+EC_Point OS2ECP(const std::vector<uint8_t, Alloc>& data, const CurveGFp& curve)
    { return OS2ECP(data.data(), data.size(), curve); }
 
-class PointGFp_Var_Point_Precompute;
+class EC_Point_Var_Point_Precompute;
+
+// The name used for this type in older versions
+typedef EC_Point PointGFp;
```

Comment:
If we want to keep supporting the old identifier, I thike it would make sense to also add a "forwarding header" `point_gfp.h` that simply includes `ec_point.h`. Maybe mark it as deprecated right from the start.